core/src/test/resources/sql/sub-query.iq

```diff
@@ -831,7 +831,7 @@ EnumerableCalc(expr#0..3=[{inputs}], expr#4=[NOT($t2)], expr#5=[IS TRUE($t4)], e
     EnumerableLimit(fetch=[1])
       EnumerableSort(sort0=[$0], dir0=[DESC])
         EnumerableAggregate(group=[{0}], c=[COUNT()])
-          EnumerableCalc(expr#0..2=[{inputs}], expr#3=[false], cs=[$t3])
+          EnumerableCalc(expr#0..2=[{inputs}], expr#3=[false], expr#4=[123], expr#5=[null:INTEGER], expr#6=[=($t4, $t5)], expr#7=[IS NULL($t5)], expr#8=[OR($t6, $t7)], cs=[$t3], $condition=[$t8])
```

Comment:
This looks like a plan degradation.
For instance `expr#4=[123], expr#5=[null:INTEGER], expr#6=[=($t4, $t5)]` is the same as `null:BOOLEAN`.

Do you know the reason for this plan degradation?

Reply:
The cost model doesn't think it is a degradation:
![image](https://user-images.githubusercontent.com/15352793/76025526-bc272500-5ef2-11ea-89a1-e04776234a55.png)

Reply:
Let me put it in another way: you change the optimizer, and now it favours bad plans.
What it does it introduces a dummy always_true filter, and it thinks the filter would reduce the number of rows and so on. It does not look like a well-behaving optimizer :-/

Even though the change reduces `slow test` execution, that reduction might be the result of "skipping some rules" rather than removing importance.

So currently it looks like some rules do not fire which result in a noticeable amount of useless predicates floating around.

Reply:
How do you know it is skipping some rules? Any evidence?

Reply:
As a human being, you know it is a bad plan, but the cost model thinks it is a better plan. Shouldn't you blame the cost model?

Reply:
OK. You propose a change. It results in generating bad plans, thus it introduces a technical regression. There's a technical justification for -1.

I know cost model has awful lot of inconsistencies. However, it turns out that all those 100 tiny inconsistencies cancel each other, and Calcite manages to produce "sane" plans.
Now you fix one or two such defects (which has good merit), however, the net result becomes that there are 98 inconsistencies in the cost model which **no longer** cancel each other.

Calcite purpose is optimizer, and it is really sad to introduce regressions to the optimizer.

Reply:
@vlsi I fixed the plan diffs as requested.